base/iostream.jl

```diff
@@ -135,6 +135,14 @@ end
 
 Seek a stream to the given position.
 
+The `pos` argument should be a value that can be returned by `position(s)`.
+Seeking before the first position should throw an error. Seeking after the
+final position should throw an error if the stream is not writable.
+If the stream is writable, reading will throw an error, and writing will fill
```

Comment:
It should be okay for `read` to return an empty vector instead of throwing an error. That is how `IOStream`, and `GZipStream` currently behave after seeking past the end. Implementing this for `skip` in `TranscodingStreams` is possible, but I think it would require keeping track of a bunch of additional states like zlib does in https://github.com/madler/zlib/blob/5a82f71ed1dfc0bec044d9702463dbdf84ea3b71/gzguts.h#L188-L196, basically making skip lazy and then running any requested skips on every read. https://github.com/madler/zlib/blob/5a82f71ed1dfc0bec044d9702463dbdf84ea3b71/gzread.c#L276-L281